ArduCopter/mode_auto.cpp

```diff
@@ -2041,4 +2041,32 @@ bool ModeAuto::verify_nav_script_time()
 }
 #endif
 
+/********************************************************************************/
+// Pause and continue auto mode
+/********************************************************************************/
+
+// pause_mission - Prevent aircraft from progressing along the track
+bool ModeAuto::pause()
+{
+    // do not pause if not in the wp_nav sub mode or already reached the wp_nav destination
+    if(!wp_nav->is_active() || wp_nav->reached_wp_destination()) {
```

Comment:
Txs for the change.

Using the wp_nav->is_active() instead of checking the submode is a good idea but it creates a opportunity for things to go wrong because "is_active()" simply checks if wp_nav's update was done with in the past 0.2sec which means that if the wp nav has just stopped being used (for example the sub mode has just changed to Circle) then we could set wp_nav->set_pause() even though it's not really being used.